### DISCUSSION TITLE
Little question about random seed for generating ensembles

### DISCUSSION THREAD

**Discussion by biglinn** [SSCHAcode/python-sscha#359] [Q&A] [answered]
Hi, @mesonepigreco:

I find that random seed is fixed in the tutorial _https://sscha.eu/Tutorials/tutorial_02_advanced_submission/_, like this:
![image](https://github.com/user-attachments/assets/ab362d1c-2fa2-4e00-ad23-56a3ae36d3aa)

This is interpreted as generating the same ensemble.
What does the `same` refer to? Distribution of displacement, or else what? And is it necessary?

**Comment by mesonepigreco** [comment, marked as answer]
"Same" means that each time the code runs, it generates the same sequence of random numbers (therefore, the same displacements). In this way, the behavior is deterministic and reproducible. It is **not** necessary; it was just employed in that tutorial as part of the SSCHA school to use an ensemble of configurations for which we had pre-computed the DFT results; in that way, everyone in the school could use the pre-computed files instead of re-running the DFT calculations on that specific ensemble.

**Reply by biglinn** [reply]
Thanks! I see it.
